src/aux-server/aux-web/shared/scene/Wall3D.ts

```diff
@@ -168,16 +179,80 @@ export class Wall3D extends Object3D {
                 this.lastSourceDir != undefined &&
                 this.sourceFile3d.position.equals(this.lastSourceDir) &&
                 this.lastScale != undefined &&
-                sourceHeight === this.lastScale
+                sourceHeight === this.lastScale &&
+                this.lastSourceWorkspace != undefined &&
+                this.lastSourceWorkspace.equals(sourceWorkspace.position) &&
+                this.lastTargetWorkspace != undefined &&
+                this.lastTargetWorkspace.equals(targetWorkspace.position)
             ) {
-                return;
+                if (this.rebuildCount > 0) {
+                    this.rebuildCount--;
+                } else {
+                    return;
+                }
             } else {
-                this.lastWidth = width;
-                this.lastDir = dir;
-                this.lastSourceDir = this.sourceFile3d.position;
-                this.lastScale = sourceHeight;
+                if (this.rebuildCount <= 0) {
+                    this.rebuildCount = 2;
+                }
             }
 
+            // double update the file's position to move the wall correctly
+            let x = this._targetFile3d.display.position.x;
+            let y = this._targetFile3d.display.position.z;
+
+            sourceWorkspace.simulation3D.simulation.helper.updateFile(
```

Comment:
@shaneYeti Is it possible to use `sourceWorkspace.simulation3D.ensureUpdate()` here instead of updating the file? Updating the file is not good for performance and might cause a bunch of issues in the future. We should find a different solution to this.